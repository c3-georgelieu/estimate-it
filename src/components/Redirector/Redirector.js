import React from 'react'
import { Redirect } from 'react-router-dom'

const Redirector = ({ pathToRedirect }) => pathToRedirect !== null
  && <Redirect to={pathToRedirect} />

export default Redirector
