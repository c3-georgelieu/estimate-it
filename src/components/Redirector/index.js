import { connect } from 'react-redux'
import Redirector from './Redirector'

const mapStateToProps = ({ redirector: { pathToRedirect } }) => ({ pathToRedirect })

export default connect(mapStateToProps, null)(Redirector)
