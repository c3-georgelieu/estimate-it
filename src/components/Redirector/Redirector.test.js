/* globals describe, it, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { HashRouter as Router } from 'react-router-dom'
import Redirector from './Redirector'

describe("Redirector's behavior", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Redirector
        redirector={{ href: '' }}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Redirector
            redirector={{ href: '/estimate/new' }}
          />
        </Router> /* eslint-disable-line comma-dangle */
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
