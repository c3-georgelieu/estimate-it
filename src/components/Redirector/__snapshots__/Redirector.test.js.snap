// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirector's behavior renders correctly 1`] = `""`;

exports[`Redirector's behavior renders correctly 2`] = `null`;
